packages/contracts-bedrock/contracts/libraries/BytesUtils.sol

```diff
@@ -3,19 +3,27 @@ pragma solidity ^0.8.9;
 
 /**
  * @title BytesUtils
+ * @notice BytesUtils is a library for manipulating byte arrays.
  */
 library BytesUtils {
-    /**********************
-     * Internal Functions *
-     **********************/
-
+    /**
+     * @custom:attribution https://github.com/GNSPS/solidity-bytes-utils/blob/6458fb2780a3092bc756e737f246be1de6d3d362/contracts/BytesLib.sol#L228-L295
+     * @notice Slices a byte array with a given starting index and length. Returns a new byte array
+     *         as opposed to a pointer to the original array. Will throw if trying to slice more
+     *         bytes than exist in the array.
+     *
+     * @param _bytes Byte array to slice.
+     * @param _start Starting index of the slice.
+     * @param _length Length of the slice.
+     *
+     * @return Slice of the input byte array.
+     */
     function slice(
         bytes memory _bytes,
         uint256 _start,
         uint256 _length
     ) internal pure returns (bytes memory) {
         require(_length + 31 >= _length, "slice_overflow");
-        require(_start + _length >= _start, "slice_overflow");
```

Comment:
Unnecessary check as of 0.8 since overflow is checked by default. Note that this is also removed in the upstream code.

Reply:
Following the same logic, the line above could be removed as well?

Reply:
Actually sorry this is incorrect, since we're using this in assembly it's actually necessary to do the checks manually. Needs to be wrapped in `unchecked` though or you'll get incorrect errors.

Reply:
Not sure exactly why that check was removed upstream but in practice you'd have to have `_start` or `_length` be close to max u256, which would use an extraordinary amount of gas. I doubt you could even get remotely close to that.

Reply:
Anyway I added the check back in just to be safe